controllers/storageclaim_controller.go

```diff
@@ -152,7 +152,7 @@ func (r *StorageClaimReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	r.storageClaimHash = getMD5Hash(r.storageClaim.Name)
 	r.storageClaim.Status.Phase = v1alpha1.StorageClaimInitializing
 
-	if r.storageClaim.Spec.StorageClient == nil {
+	if r.storageClaim.Spec.StorageClientName == nil {
```

Comment:
When working with string it is preferable to check against the empty string
```suggestion
        if r.storageClaim.Spec.StorageClientName == "" {
```

Reply:
It is a pointer atm.

Reply:
This is confusing. Is it the pointer that is nill or the string that is nil (the storage container inside the string)? 

Of course there is only one correct answer:
`r.storageClaim.Spec.StorageClientName == nil` the pointer is nil
`*r.storageClaim.Spec.StorageClientName == nil` the string is nil

But I hope you do understand why it is so confusing. As mentioned in another comment, I don't see the value of having a pointer to a string in the first place, and this is a good example why we should not add unnecessary complexity

Reply:
I don't understand, a pointer to string can be nil but a string can't be nil.

Reply:
You are correct a string var cannot be assigned nil, I was thinking about slices. But regardless empty string will serve us exactly like nil. And as empty string is the default value for a string it means that if the user does no provide a value we will get an empty string in the field.

Reply:
done, change to string.